tls/s2n_psk.c

```diff
@@ -99,6 +99,25 @@ S2N_RESULT s2n_psk_clone(struct s2n_psk *new_psk, struct s2n_psk *original_psk)
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_resumption_psks_remove(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+
+    struct s2n_array *psk_list = &conn->psk_params.psk_list;
+    for (uint32_t i = psk_list->len; i > 0; i--) {
+        /* This index variable along with the loop condition keep size_t i from underflowing */
+        size_t idx = i - 1;
+        struct s2n_psk *psk = NULL;
+        RESULT_GUARD(s2n_array_get(psk_list, idx, (void**)&psk));
+        RESULT_ENSURE_REF(psk);
+
+        if (psk->type == S2N_PSK_TYPE_RESUMPTION) {
```

Comment:
Is there a possibility of the psk_list containing both external and resumption type PSKs in the same connection? In the real world use-case we would either have external PSKs or resumption PSKs here.

In which case, as this function is to remove only the resumption PSKs,  should we error/return when `psk->type != S2N_PSK_TYPE_RESUMPTION`?

Reply:
We don't want to let people have both resumption and external psks, but I think that if we want that kind of error checking it should be in the add psks methods rather than the remove psks methods.

Reply:
In this method we should be freeing PSKs with `S2N_PSK_TYPE_RESUMPTION` type only,  if we come across external PSK using this remove function, it is not correct. The current choice is to silently ignore, my question is whether we should error instead?

Reply:
We have now changed this pr to free all psks, not just resumption psks, as a customer adding a resumption ticket is indicating that they are not interested in using their previously-set external psks.

Reply:
We actually currently can have both types. However, we do have a future task to change that, so this this change is fine: https://github.com/aws/s2n-tls/issues/2685